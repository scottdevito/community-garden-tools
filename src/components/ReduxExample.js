import React, { Component } from 'react';
import styled from 'styled-components';

import ResourcesHOC from './reusable/ResourcesHOC';
import { Card, CTAPrimaryBlue } from '../reusable-styles/reusable-styles';
import SignalRExample from './SignalRExample';

class ReduxExample extends Component {
  handleActionClick = () => {
    this.props.getAPIDataPromise();
  };

  render() {
    const { exampleState, signalRState } = this.props;
    let reduxExmapleSection = null;

    if (exampleState && exampleState !== []) {
      reduxExmapleSection = `Data: ${exampleState.title}`;
    }

    return (
      <ReduxExampleWrapper>
        <h1>Redux</h1>
        <ul>
          <li>npm i --save redux react-redux redux-promise redux-thunk</li>
          <li>Add redux specific code into src/index.js</li>
        </ul>
        <ResourcesHOC>Some Resource</ResourcesHOC>

        <h3>Test Data Fetch Action:</h3>
        <CTAPrimaryBlue onClick={() => this.handleActionClick()}>
          Dispatch Redux Action
        </CTAPrimaryBlue>
        {reduxExmapleSection}
        <SignalRExample
          sendSocketMessage={this.props.sendSocketMessage}
          message={signalRState.message}
          notification={signalRState.notification}
        />
      </ReduxExampleWrapper>
    );
  }
}

export default ReduxExample;

const ReduxExampleWrapper = styled(Card)`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;
