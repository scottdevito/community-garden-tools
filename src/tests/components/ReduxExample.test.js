import React from 'react';
import ReactDOM from 'react-dom';
import ReduxExample from '../../components/ReduxExample';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';

// Snapshot test
it('renders correctly', () => {
  const tree = renderer.create(<ReduxExample />).toJSON();
  expect(tree).toMatchSnapshot();
});
