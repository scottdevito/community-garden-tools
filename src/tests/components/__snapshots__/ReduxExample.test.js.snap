// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bZQynM dETavn"
>
  <h1>
    Redux
  </h1>
  <ul>
    <li>
      npm i --save redux react-redux redux-promise redux-thunk
    </li>
    <li>
      Add redux specific code into src/index.js
    </li>
  </ul>
  <div
    className="sc-ifAKCX eVrQyi"
  >
    <h2>
      Resources:
    </h2>
    Some Resource
  </div>
  <h3>
    Test Data Fetch Action:
  </h3>
  <button
    className="sc-bxivhb iBHaDI"
    onClick={[Function]}
  >
    Dispatch Redux Action
  </button>
  <div
    className="sc-EHOje enDNEB"
  >
    <div>
      Send a SignalR Message.
    </div>
    <i>
      Note: This will only work while connected to SignalR API.
    </i>
    <br />
    <button
      onClick={[Function]}
    >
      Send message
    </button>
    <div>
      Response: no signal yet
    </div>
    <br />
    <div>
      Get all Inventory to recieve notification: no message yet
    </div>
  </div>
</div>
`;
